### PR TITLE
Pose update (with per-measurement covariance) from Odometry message

### PR DESCRIPTION
**Pull request by nicolov**
This PR adds PoseUpdate input from nav_msgs::Odometry messages.

When using GPS, it's useful to correct the update covariance depending on the value reported by the receiver. To do this, this PR adds optional scaling of the configured pose update covariance by the value in the odometry message. This behavior is controlled by the new `useOdometryPoseCov` flag in the .info file.

Notes:
- I've decided to keep the configured value as a scaling factor to make "optimistic" or "pessimistic" receivers easier to handle.
- I've only added support for scaling position covariance, for the common case of GPS. I can add a second flag for attitude if needed.

@mhkabir

### CODE REVIEW

**Comment by bloesch** [comment]
ok to test

File: include/rovio/PoseUpdate.hpp [hunk 1]
```diff
@@ -60,8 +60,10 @@ class PoseUpdateMeas: public LWF::State<LWF::VectorElement<3>,LWF::QuaternionEle
  public:
   static constexpr unsigned int _pos = 0;
   static constexpr unsigned int _att = _pos+1;
+  M3D pos_cov_; // Will be used to scale the update covariance according to the measurement
```
Comment (bloesch) [inline include/rovio/PoseUpdate.hpp]: I think it would be better to use the full 6x6 matrix. More generic and still easy to use.

Comment (nicolov) [inline include/rovio/PoseUpdate.hpp]: I'd add another flag for the other 3x3 block though, do you agree? For
example, my gps drivers sets the orientation diagonal to -1, so I wouldn't
want to use that..

Comment (bloesch) [inline include/rovio/PoseUpdate.hpp]: The 6x6 matrix would also consider cross-correlations between position and attitude. But I see the problem with the potentially invalid data. What about making the reading in the callback (RovioNode.hpp) dependent on enablePosition_/enableAttitude_?

Comment (nicolov) [inline include/rovio/PoseUpdate.hpp]: Ok, it makes sense. It would be a global toggle + the two existing ones for
position and attitude.

I've thought about non-linearities in the innovation covariance inversion
step (`Py_`) and believe that the off-diagonal blocks should be zeroed out
if either position or attitude is set to be unused. This is not a problem
in the current code because `updnoiP_` is diagonal by construction, but it
would be if we allow arbitrary 6x6 covariance matrices. Do you agree?

In other words, the current code relies on the fact that the inverse of a
block-diagonal matrix is still a block-diagonal matrix where the inversion
can be done blockwise. This is not true in the general case we want to
allow here ([reference](
https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_inversion).

Comment (nicolov) [inline include/rovio/PoseUpdate.hpp]: Ok, it makes sense. It would be a global toggle + the two existing ones for position and attitude.

I've thought about non-linearities in the innovation covariance inversion
step (`Py_`) and believe that the off-diagonal blocks should be zeroed out
if either position or attitude is set to be unused. This is not a problem
in the current code because `updnoiP_` is diagonal by construction, but it
would become so if we allow arbitrary 6x6 covariance matrices. Do you agree?

In other words, the current code relies on the fact that the inverse of a
block-diagonal matrix is still a block-diagonal matrix and the inversion
can be done blockwise. This is not true in the general case we want to
allow here ([reference](
https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_inversion)).

Comment (bloesch) [inline include/rovio/PoseUpdate.hpp]: I agree that off-diagonal blocks become an issue if either position or attitude is disabled. But those blocks can easily be set to zero in such a case. Do I miss something?

**Comment by bloesch** [comment]
lgtm. thanks!

**Comment by mhkabir** [comment]
@nicolov I just ported this over into my working branch for tests. In what form should we send the unknown covariance components (e.g inflated to a large value) ?

**Comment by nicolov** [comment]
Yep, that's right. A lot of ROS code seems to use -1 to mark such invalid values, so watch out for that
